src/puya/awst/nodes.py

```diff
@@ -1475,6 +1776,14 @@ def accept(self, visitor: ExpressionVisitor[T]) -> T:
 
 @attrs.frozen
 class Enumeration(Expression):
+    """
+    Returns an iterator-like expression which can be used in a ForInLoop. The iterator emits a
```

Comment:
Similar vibe to `Range` here, it's a lazy-sequence, and only usable with `ForInLoop`.